src/main/java/com/robertx22/mine_and_slash/a_libraries/jei/RunewordCategory.java

```diff
@@ -0,0 +1,108 @@
+package com.robertx22.mine_and_slash.a_libraries.jei;
+
+import com.robertx22.mine_and_slash.database.data.runewords.RunewordRecipe;
+import com.robertx22.mine_and_slash.database.registry.ExileDB;
+import com.robertx22.mine_and_slash.mmorpg.registers.common.items.RuneItems;
+import com.robertx22.mine_and_slash.uncommon.localization.Words;
+import mezz.jei.api.gui.builder.IRecipeLayoutBuilder;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.IFocusGroup;
+import mezz.jei.api.recipe.RecipeType;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RunewordCategory implements IRecipeCategory<RunewordRecipe> {
+    final IDrawable background;
+    IGuiHelper guiHelper;
+    RecipeType<RunewordRecipe> recipeType;
+    final IDrawable icon;
+
+    RunewordExtension recipeExtension;
+
+    RunewordCategory(IGuiHelper guiHelper, RecipeType<RunewordRecipe> type) {
+        this.guiHelper = guiHelper;
+        this.recipeType = type;
+        icon = guiHelper.createDrawableItemStack(RuneItems.MAP.entrySet().stream().findFirst().get().getValue().get().getDefaultInstance());
+
+        Dimensions dimension = calculateDimension();
+        int maxWidth = dimension.width;
+        int maxLines = dimension.lines;
+
+        recipeExtension = new RunewordExtension(maxWidth);
+        var height = recipeExtension.calculateMaxHeight(maxLines);
+        background = guiHelper.createBlankDrawable(maxWidth, height);
```

Comment:
While with plain JEI in goes over the bounds of UI, EMI handles big size well
<img width="953" height="854" alt="image" src="https://github.com/user-attachments/assets/a2cba09e-1884-4241-8795-611e05e5c4ec" />